nsxt/resource_nsxt_policy_project.go

```diff
@@ -0,0 +1,211 @@
+	displayName := d.Get("display_name").(string)
+	description := d.Get("description").(string)
+	tags := getPolicyTagsFromSchema(d)
+	shortID := d.Get("short_id").(string)
+	siteInfosList := d.Get("site_info").([]interface{})
+	var siteInfos []model.SiteInfo
+	for _, item := range siteInfosList {
+		data := item.(map[string]interface{})
+		var edgeClusterPaths []string
+		for _, ecp := range data["edge_cluster_paths"].([]interface{}) {
+			edgeClusterPaths = append(edgeClusterPaths, ecp.(string))
+		}
+		sitePath := data["site_path"].(string)
+		obj := model.SiteInfo{
+			EdgeClusterPaths: edgeClusterPaths,
+			SitePath:         &sitePath,
+		}
+		siteInfos = append(siteInfos, obj)
+	}
+	var tier0s []string
+	for _, t0 := range d.Get("tier0_gateway_paths").([]interface{}) {
+		t := t0.(string)
+		tier0s = append(tier0s, t)
+	}
+
+	obj := model.Project{
+		DisplayName: &displayName,
+		Description: &description,
+		Tags:        tags,
+		SiteInfos:   siteInfos,
+		Tier0s:      tier0s,
+	}
+
+	if shortID != "" {
+		obj.ShortId = &shortID
+	}
+
+	log.Printf("[INFO] Patching Project with ID %s", id)
+
+	client := infra.NewProjectsClient(connector)
+	return client.Patch(defaultOrgID, id, obj)
+}
+
+func resourceNsxtPolicyProjectCreate(d *schema.ResourceData, m interface{}) error {
+
+	// Initialize resource Id and verify this ID is not yet used
+	id, err := getOrGenerateID(d, m, resourceNsxtPolicyProjectExists)
+	if err != nil {
+		return err
+	}
+
+	connector := getPolicyConnector(m)
+	err = resourceNsxtPolicyProjectPatch(connector, d, m, id)
+	if err != nil {
+		return handleCreateError("Project", id, err)
+	}
+
+	d.SetId(id)
+	d.Set("nsx_id", id)
+
+	return resourceNsxtPolicyProjectRead(d, m)
+}
+
+func resourceNsxtPolicyProjectRead(d *schema.ResourceData, m interface{}) error {
+	connector := getPolicyConnector(m)
+
+	id := d.Id()
+	if id == "" {
+		return fmt.Errorf("Error obtaining Project ID")
+	}
+
+	var obj model.Project
+	client := infra.NewProjectsClient(connector)
+	var err error
+	obj, err = client.Get(defaultOrgID, id)
+	if err != nil {
+		return handleReadError(d, "Project", id, err)
+	}
+
+	d.Set("display_name", obj.DisplayName)
+	d.Set("description", obj.Description)
+	setPolicyTagsInSchema(d, obj.Tags)
+	d.Set("nsx_id", id)
+	d.Set("path", obj.Path)
+	d.Set("revision", obj.Revision)
+
+	d.Set("short_id", obj.ShortId)
+	var siteInfosList []map[string]interface{}
+	for _, item := range obj.SiteInfos {
+		data := make(map[string]interface{})
+		data["edge_cluster_paths"] = item.EdgeClusterPaths
+		data["site_path"] = item.SitePath
+		siteInfosList = append(siteInfosList, data)
+	}
+	d.Set("site_info", siteInfosList)
+	d.Set("tier0_gateway_paths", obj.Tier0s)
+
+	return nil
+}
+
+func resourceNsxtPolicyProjectUpdate(d *schema.ResourceData, m interface{}) error {
+
+	id := d.Id()
+	if id == "" {
+		return fmt.Errorf("Error obtaining Project ID")
+	}
+
+	headers := map[string]string{"enable_partial_patch": "true"}
```

Comment:
Is this header still needed?

Reply:
In theory we do: I've added this to handle that issue, where NSX was overriding short_id on update.
The NSX bug is fixed but there are still released NSX versions which support projects but do not have this fix.
However - tests seem to pass without it - so I'll remove it :)

Reply:
We'll officially claim multitenancy support starting from next release, so the fix will be in

Reply:
Cool, I'll leave `getPolicyConnectorWithHeaders()` just in case we'll even need it?